tools/backups2datalad/util.py

```diff
@@ -397,3 +422,126 @@ def exp_wait(
     while attempts is None or n < attempts:
         yield base ** n * multiplier
         n += 1
+
+
+async def arequest(client: httpx.AsyncClient, method: str, url: str) -> httpx.Response:
+    waits = exp_wait(attempts=5, base=1.25, multiplier=1.25)
+    while True:
+        try:
+            r = await client.request(method, url, follow_redirects=True)
+            r.raise_for_status()
+        except httpx.HTTPError as e:
+            if isinstance(e, httpx.RequestError) or (
+                isinstance(e, httpx.HTTPStatusError) and e.response.status_code >= 500
+            ):
+                try:
+                    delay = next(waits)
+                except StopIteration:
+                    raise e
+                log.warning(
+                    "Retrying %s request to %s in %f seconds as it raised %s: %s",
+                    method.upper(),
+                    url,
+                    delay,
+                    type(e).__name__,
+                    str(e),
+                )
+                await trio.sleep(delay)
+                continue
+            else:
+                raise
+        return r
+
+
+@dataclass
+class Remote:
+    name: str
+    prefix: str
+    uuid: str
+
+
+def init_dataset(
+    ds: Dataset,
+    desc: str,
+    commit_date: datetime,
+    backup_remote: Optional[Remote] = None,
+    backend: str = "SHA256E",
+    cfg_proc: Optional[str] = "text2git",
+) -> None:
+    log.info("Creating dataset for %s", desc)
+    datalad.cfg.set("datalad.repo.backend", backend, where="override")
```

Comment:
lets place it within `try/finally` with `datalad.cfg.unset("datalad.repo.backend", where="override")` in finally, so we do not carry this side-effect into possibly other relevant invocations.  

And I forgot (and failed to search up a hit in this repo) why we have not used `ds.repo.set_default_backend(backend, persistent=True)` after `ds.create`.  May be @jwodder finds a trace of why we didn't go that way?

Reply:
* Done.
* I was not aware of the latter option.